auth/src/main/java/org/aerogear/auth/impl/GuestAuthenticatorImpl.java

```diff
@@ -22,21 +25,20 @@
     /**
      * Roles to be assigned to the guest user
      */
-    private final IRole[] roles;
+    private final Map<RoleKey, IRole> roles;
 
     /**
      * Builds a new guest authenticator object
      * @param guestUser the user to be returned after the 'authentication'
      * @param roles the roles to be assigned to the user
      */
-    public GuestAuthenticatorImpl(final String guestUser, final IRole[] roles) {
+    public GuestAuthenticatorImpl(final String guestUser, final Map<RoleKey, IRole> roles) {
```

Comment:
I think an array or collection of roles as parameter would be better. The map can be constructed inside the class.